src/CSV.jl

```diff
@@ -1028,6 +1058,8 @@ Parses a delimited file into a `DataFrame`. `copycols` determines whether a copy
 """
 read(source; copycols::Bool=false, kwargs...) = DataFrame(CSV.File(source; kwargs...), copycols=copycols)
 
+DataFrames.DataFrame(f::CSV.File; copycols::Bool=false) = DataFrame(getcolumns(f), getnames(f); copycols=copycols)
```

Comment:
This is a slightly unfortunate fallout of `CSV.File <: AbstractVector`, since the fallback `DataFrame` constructor special cases that check and it fails for the case when `CSV.File` has zero rows.

Reply:
It's unfortunate that you need to define this at all, you mean?

Reply:
Do you think something can be improved in DataFrames? Otherwise, doesn't sound too bad.

Reply:
I wouldn’t say it’s too bad, either. But of course, the best interfaces support blind composition... so I guess we could look at the constructors at some point?

Reply:
Woops, something I missed is that `copycols` should definitely default to `true` here, for consistency with the generic method defined in DataFrames. Also this PR is breaking because of that: people who did `df = DataFrame(CSV.File(...)); append!(df, row)` will now get an error, but previously it worked. I've filed https://github.com/JuliaData/CSV.jl/pull/531.